### PR TITLE
Add dryrun transaction API

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add ain_dryrunSignedTransaction json rpc API
- Update transfer tool for dryrun
- Add test for dryrun transaction API 
- Simplify db restoring logic

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/1182

### CODE REVIEW

File: node/index.js [hunk 1]
```diff
@@ -792,8 +792,10 @@ class BlockchainNode {
...
           `(${perAccountFreePoolSize}) for account: ${executableTx.address}`);
     }
+    const eventSource = isDryrun ? null : ValueChangedEventSources.USER;
     const result = this.db.executeTransaction(
-        executableTx, false, true, this.bc.lastBlockNumber() + 1, this.bc.lastBlockTimestamp(), ValueChangedEventSources.USER);
+        executableTx, false, true, this.bc.lastBlockNumber() + 1, this.bc.lastBlockTimestamp(),
+        eventSource, isDryrun);
     if (CommonUtil.isFailedTx(result)) {
       if (DevFlags.enableRichTransactionLogging) {
         logger.error(
```
Comment (shyun-comcom) [inline node/index.js:801]: minor) Is it correct to log this even for the case of `!isDryrun`?

File: p2p/server.js [hunk 2]
```diff
@@ -818,6 +818,7 @@ class P2pServer {
 
           continue;
         }
+        // NOTE(platfowner): There is no dryrun mode for batch transactions.
```
Comment (shyun-comcom) [inline p2p/server.js:821]: Returns error when `isDryrun` is true for batch transaction?

File: node/index.js (hunk 1 above)
Comment (platfowner) [reply node/index.js:801]: enableRichTransactionLogging flag is used for debugging. so it's just fine with the error logging.

File: p2p/server.js (hunk 2 above)
Comment (platfowner) [reply p2p/server.js]: Just changed to allow dryrun for batch txs in executeAndBroadcastTransaction().
Note however that actually it's impossible to send batch txs with isDryrun = true as we don't support dryrunSignedTransactionBatch().

**Comment by platfowner** [comment]
PTAL~ @shyun-comcom

**Comment by shyun-comcom** [comment]
Function `createAndExecuteTransaction` calls `executeAndBroadcastTransaction`.
Is there no need to set the `isDryrun` flag [here](https://github.com/ainblockchain/ain-blockchain/blob/feature/platfowner/feature/client/index.js#L849)?

**Comment by platfowner** [comment]
> Function `createAndExecuteTransaction` calls `executeAndBroadcastTransaction`. Is there no need to set the `isDryrun` flag [here](https://github.com/ainblockchain/ain-blockchain/blob/feature/platfowner/feature/client/index.js#L849)?

createAndExecuteTransaction is for creating transactions for Dev Client SET APIs. Here are some context:
- Dev Client SET APIs are used for testing and development purposes and being disabled in testnet and mainnet deployment
- We have some plan to deprecate Dev Client SET APIs once all the test cases are migrated to json rpc APIs (there so many tests cases and it will take some time) 
- Currently isDryrun = true option is available for json rpc API ain_sendSignedTransactionDryrun only

**Review by shyun-comcom — APPROVED** [review]
LGTM!

**Pull request merged**